.ci.yaml

```diff
@@ -2387,12 +2387,14 @@ targets:
   # linux motog4 benchmark
   - name: Linux_android hello_world__memory
     recipe: devicelab/devicelab_drone
-    presubmit: false
```

Comment:
We need to be careful when enabling the test in pre-submit. There is currently only one linux/motoG4 testbed in presubmit. If it fails, it will block PRs.

@reidbaker Is this target suitable to be migrated to linux/pixel, which we have more capacity? BTW: we are trying to migration motoG4 to pixel considering motoG4 is dying and more difficult to maintain.
@yusuf-goog If the above answer is no, do we still have the linux/motoG4 to make it as least 2 testbeds in try?

Reply:
We should not be moving any tests specifically to moto G4. They are to be considered end of life. Any tests requiring specific low in device hardware should move to use the A02s devices. All other tests should be setup to choose either a pixel or motoG4 as available.

Additionally, devicelab builds are not meant to run in try currently. As Keyong mentioned, we would need more testbeds to support blocking (builds that might turn the tree red). The current testbeds in `try` are meant to run experimental (non-blocking) builds.

Reply:
@keyonghan @yusuf-goog 
If you look at the run if it triggers only when the hello world code is modified. If this does not run then the code being modified is, to my knowledge, not tested. The issue I am fixing is one where we submitted a change that broke the tree because we modified this example and it did not compile. 

Here is the commit history for the main build file. It has seen ~18 changes since the end of 2019, 5 of which were reverts. This is an infrequently changed part of our system. 
https://github.com/flutter/flutter/commits/c576f0039d75716f04f82c8b58ca208c756b9835/examples/hello_world/android/app/build.gradle

As to if this test can be run on another device potentially it can be run on another device but that would change the baseline that performance is measured against and I believe is out of scope for this bug. 

What needs to happen to make the 2 of you happy with this pr and for us to get the presubmit protection for the tree?

Reply:
It is really an issue with capacity :neutral_face: FWIW: the team has been working hard to migrate tests off of MotoG4: https://github.com/flutter/flutter/issues/140893

This test will be triggered in presubmit for any PR with a `.ci.yaml` change as well. All `presubmit: true` targets will be triggered for a `.ci.yaml` change, and this will be a lot. Example: all those flake/deflake PRs.

If this test is feasible to run on a pixel, I would recommend to switch it now (IIRC, the performance baseline will be re-built within 7 commits). With this switched to a pixel, you can enable it with `presubmit: true` considering we have enough capacity in the try pool and we will get the presubmit protection.

Reply:
I didnt need/want it to trigger on ci.yaml changes but had to in c33af9843fa8f2c4e1cd057ffd9f33a9ca410749 because it is enforced. Is there another way to trigger this test on code changes to the example but otherwise only run the test in post submit? 

@dnfield Do you know who might care the most about this test and if it is ok to migrate to a different device?

Reply:
You are on the correct path by using the `runIf`.
All targets are triggered considering the .ci.yaml change may be on platform level, which would potentially affect a wider portion of targets. To improve resource usage, we need to have the scheduler be smarter: https://github.com/flutter/flutter/issues/134697 (which to trigger only affected targets).

The best way moving forward is use a `pixel` phone if it runs okay.

In the long term, if this test doesn't care about device, we can support a new platform category which can run on either device. (Similar to `Mac` compared to `Mac_x64`+`Mac_arm64`):
`Linux_android` v.s. `Linux_pixel`+`Linux_motoG4`+`Linux_samsung`.

Reply:
Since it is a benchmark test I think that the device does matter. 

It sounds like in order to get this test coverage I either need to migrate the benchmark test or write a new test that is hardware independent that gives me the coverage I want.